lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -397,6 +397,48 @@ class AwsCompileFunctions {
         delete functionResource.Properties.VpcConfig;
       }
 
+      const fileSystemConfigs = functionObject.fileSystemConfigs;
+
+      if (Array.isArray(fileSystemConfigs) && fileSystemConfigs.length === 1) {
```

Comment:
At this point, as we're backed by schema validation, I would just assume that if `fileSystemConfigs` it is an array, and has at most length 1.

Therefore I'd write it as `if (fileSystemConfigs.length)`

Reply:
I still kept the `Array.isArray` check here, because if I understand it correctly, schema is not enforced at the moment and for example in tests, with missing `fileSystemConfigs` it's erroring out as it tries to check length of `undefined`. Please let me know what are your thoughts on this

Reply:
> schema is not enforced at the moment

That's true, but we also live with an agreement, that if schema is not satisfied then things can break badly. Code is not bulletproof to accept any kind of input in config, and may crash with non informative programmer errors (like `x.x.x is undefined`) - and it was always like that.

Also with one of the major upgrades we will crash by default when schema is not satisfied (warn by default is transitory state)

Therefore to simplify maintenance let's blindly assume that at this point configuration adheres to schema. I believe it's totally fine, and let us deal with less footprint